### ISSUE TITLE
Support non-struct types

### ISSUE THREAD

**Issue by philhofer** [tinylib/msgp#61]
Things get rather difficult if we *don't* want to generate code for arbitrary types, but still want to support something like:

```go
type Chunk [128]byte

type Block struct {
    Meta string
    Data Chunk
}
```

Right now, `Chunk` is interpreted as `IDENT`, even though we could conceivably generate the appropriate code for it in-situ. However, generating the appropriate methods for `Chunk` is simpler than transitively applying its type information across every location in which it is used.

TL;DR generate code for all valid `*ast.TypeDecl` nodes.

**Comment by philhofer** [comment]
See #60

**Comment by philhofer** [comment]
Fixed by #60
